libnavigation-core/build.gradle

```diff
@@ -15,6 +15,7 @@ android {
     compileSdkVersion androidVersions.compileSdkVersion
 
     compileOptions {
+        coreLibraryDesugaringEnabled = true
```

Comment:
turned on desugaring to be able to work with `LocalDateTime`

Reply:
TBH, I'm not sure that it's worth using desugaring just to simplify time parsing and usage. I haven't measured exact price of desugaring yet. My expectations is that it increases a bit binary size and compilation time. 
@mapbox/navigation-android , what's your opinion about it? If there're no strong "No" opinions, I will measure compilation time and binary size increase.

Reply:
[You can run mobile metrics](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/18269/workflows/5aba3375-583f-4eb4-b396-a030c3cc6b33) for your pull request and lookup the differences in binary size. It would be nice to have better reporting, but the pipeline is set up to answer these types of questions

![Screen Shot 2022-05-23 at 11 42 42 AM](https://user-images.githubusercontent.com/3021882/169885689-a31e5cb8-e32e-41e4-a91a-67f49b655916.png)

Reply:
| branch  | size of the test app (bytes) | size of core module (bytes) |
| ------------- | ------------- | - |
| main (09cd530425)  | 38027797  | 579796 |
| vv-invalidate-traffic-annotations (current)  | 38032865  | 600117 |
| diff | + 5068  | + 20321  |

+20 kb in core module. Doesn't seem much. Anyway it's strange that test app didn't increased size as core module, R8 is disabled 🤔

Reply:
With respect to resolving and comparing time zones, I'm now sure that `ZonedDateTime` is worth +20kb

Reply:
It turned out that desuraring doesn't work for library modules, only for applications https://issuetracker.google.com/issues/202868795.
I don't think it's a good idea to ask our customers to turn on desugaring for theirs applications, so I will turn desugaring off and get rid of `ZonedDateTime`

Reply:
wow, interesting

Reply:
Now the feature uses Java 7 `Date`